src/katello/agent/katelloplugin.py

```diff
@@ -154,7 +154,22 @@ def update_settings():
     """
     rhsm_conf = Config(RHSM_CONFIG_PATH)
     certificate = ConsumerIdentity.read()
-    plugin.cfg.messaging.cacert = rhsm_conf['rhsm']['repo_ca_cert'] % rhsm_conf['rhsm']
+    ca_cert_dir = rhsm_conf['rhsm']['ca_cert_dir']
+    # the 'katello-default-ca.pem' is the ca used for generating the CA certs.
+    # the 'candlepin-local.pem' is there for compatibility reasons (the old path where the
+    # legacy installer was putting this file. If none of them is present, there is still
+    # a chance the rhsm_conf['rhsm']['repo_ca_cert'] is serving as the CA for issuing
+    # the client certs
+    ca_candidates = [ca_cert_dir + 'katello-default-ca.pem', ca_cert_dir + 'candlepin-local.pem', rhsm_conf['rhsm']['repo_ca_cert'] % rhsm_conf['rhsm']]
```

Comment:
Using `os.path.join()` instead of (+) would be safer.  The value of *ca_cert_dir* may not end with a '/'.